### PR TITLE
check every 100ms for DFU mode

### PR DESCRIPTION
**Pull request by ctzsnooze**
Rather than wait a full 5s to flash on my Mac, this code checks for a DFU connection every 100ms, then flashes when DFU is available.

Huge THANK YOU to @haslinghuis for re-factoring my original code so well, and for @Limon for showing me the basics of how to do it!  Very grateful for your help.

On my mac, most DFU connections take around 1.6 - 1.8s.  Keen to hear how long it takes on Linux / Windows.

Typical console log at connect time.:

```
serial: closed connection with ID: 1, Sent: 19 bytes, Received: 77 bytes
port_handler.js:129 PortHandler - Removed: [{"path":"/dev/cu.usbmodem0x80000001","displayName":"Betaflight STM32F411","vendorId":1155,"productId":22336},{"path":"/dev/tty.usbmodem0x80000001","displayName":"Betaflight STM32F411","vendorId":1155,"productId":22336}]
stm32.js:121 DFU available after 1.8 seconds
stm32usbdfu.js:98 USB DFU detected with ID: 0
stm32usbdfu.js:124 Device opened with Handle ID: 1
```

We could check only every second, I guess, but it's interesting to see how quickly different systems connect.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by haslinghuis — DISMISSED** [review]

**Comment by ctzsnooze** [comment]
Thanks Mark @haslinghuis ! 

That's a much nicer way to code it.  Really grateful.  

Works great on all boards I've tried so far (F722, F405, F411).

**Review by haslinghuis — DISMISSED** [review]

File: src/js/protocols/stm32.js
Comment (Asizon) [inline src/js/protocols/stm32.js:135]: Why define after use it into the function?

File: src/js/protocols/stm32.js
Comment (Asizon) [inline src/js/protocols/stm32.js:136]: Same here

**Review by asizon — CHANGES_REQUESTED** [review]

File: src/js/protocols/stm32.js
Comment (haslinghuis) [reply src/js/protocols/stm32.js:135]: It's actually called in the setInterval below. The function definition is above the call. So need for hoisting.

File: src/js/protocols/stm32.js
Comment (haslinghuis) [reply src/js/protocols/stm32.js:136]: Same as above.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by limonspb — DISMISSED** [review]
nice!
looks good, but if there will be some change requests, plz rename
waitForDFU -> waitForDfu

**Review by limonspb — APPROVED** [review]

**Review by asizon — APPROVED** [review]

**Review by haslinghuis — APPROVED** [review]

**Pull request merged**
